drivers/display/ssd16xx.c

```diff
@@ -974,6 +975,44 @@ static int ssd16xx_init(const struct device *dev)
 	return ssd16xx_controller_init(dev);
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int ssd16xx_pm_action(const struct device *dev,
+			     enum pm_device_action action)
+{
+	int err = 0;
+	const struct ssd16xx_config *config = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		err = ssd16xx_controller_init(dev);
+		if (err < 0) {
+			return err;
+		}
+		break;
+
+	case PM_DEVICE_ACTION_SUSPEND:
+		ssd16xx_busy_wait(dev);
```

Comment:
You might need to call `ssd16xx_activate(dev, SSD16XX_CTRL2_ENABLE_CLK);` to avoid deadlocking if the logic clock hasn't been enabled here.